src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -411,8 +414,14 @@ private static void SplitCounterGroup(DocumentsOperationContext context, Collect
                     tvb.Add(fst.BasePointer, fst.Size);
                     tvb.Add(collectionSlice);
                     tvb.Add(context.TransactionMarkerOffset);
-                    table.Update(storageId, tvb);
-
+                    if (storageId.HasValue)
+                    {
+                        table.Update(storageId.Value, tvb);
```

Comment:
Why do you need that?

Reply:
when I call `SplitCounterGroup` from `PutCounters`  I do not have the table value reader at hand anymore (because now I do that in the end, after the loop). 
when I call `SplitCounterGroup` from `IncrementCounter` I do have the value reader

Reply:
Yes, but why does this matter? Shouldn't we just always call `Set` ?